### ISSUE TITLE
[Accessibility] Tuplets skipped by Alt+Left and Alt+Right

### ISSUE THREAD

**Issue by shoogle** [musescore/MuseScore#17600]
### Issue type

UI bug

### Bug description

The `next-element` (Alt+Right) and `previous-element` (Alt+Left) actions are able to select notes within a tuplet, however the tuplet itself cannot be selected.

### Steps to reproduce

1. Create a score
2. Select a rest and press Ctrl+3 to add a tuplet.
3. Select a nearby note or rest.
4. Try to navigate to the tuplet with Alt+Left or Alt+Right.

### Screenshots/Screen recordings

![image](https://github.com/musescore/MuseScore/assets/11011881/eda1e5c6-404e-435e-af7c-c91877d12d7c)

### MuseScore Version

Latest nightly (231340304, 0e5a099)

### Regression

No.

### Operating system

Windows 11

**Comment by codecruisedor** [comment]
Hello,
This is my first time attempting to contribute to an issue.

I would like to contribute to this issue..but I have a few questions and would really appreciate some guidance. What key combination would be appropriate for selecting a tuple? Also, what would be the general flow programmatically from pressing the key combination to a selection of the tuple?

Thank you.

**Comment by MarcSabatella** [comment]
The shortcuts to select most elements are Alt+Left/Right.  Those should move through scores element by element.  The problem is they skip the tuplet bracket.  Presumably it should be hit just after the first note of the tuplet when traversing the score with Alt+Right.  You would need to discover where this is handled and make sure the bracket is included.

**Comment by shoogle** [comment]
@codecruisedor, sure.

Let's say you start with the C selected at the beginning of the measure in this example, and then press Alt+Right repeatedly.

![image](https://github.com/musescore/MuseScore/assets/11011881/ab343f71-e943-4b32-9872-f77046fe6a8a)

Here's what should be selected after each press of Alt+Right:

1. Note D
2. The tuplet
3. Note E
4. Note F
5. Note G
6. Note A

That's what happens already except the tuplet is skipped.

If you go in the other direction with Alt+Left then the elements should be selected in reverse order.

**Comment by shoogle** [comment]
If you [check in shortcuts.xml](https://github.com/musescore/MuseScore/blob/8288f7dbf8bf1ee9ee7c378c991efeffc3d24053/src/framework/shortcuts/data/shortcuts.xml#L382-L389), you'll see that the Alt+Right shortcut is mapped to an action called `next-element`. If you [search the source code](https://github.com/musescore/MuseScore/wiki/Find-your-way-around-the-code) for the string `"next-element"` then it will give you an entry-point in the program from which you can work backwards until you find the logic that you need to change.

For example, searching for `"next-element"` takes you to [this line in notationactioncontroller.cpp](https://github.com/musescore/MuseScore/blob/8288f7dbf8bf1ee9ee7c378c991efeffc3d24053/src/notation/internal/notationactioncontroller.cpp#L187-L188). That line mentions a function called `registerMoveSelectionAction()`, which is defined [here](https://github.com/musescore/MuseScore/blob/8288f7dbf8bf1ee9ee7c378c991efeffc3d24053/src/notation/internal/notationactioncontroller.cpp#L2051-L2068). The function calls another function called `moveSelection()`, which......

I'll save you the effort and say that you're looking for functions called `nextElement()` and `previousElement()`. These are virtual functions that all EngravingItems can have. You need to make sure that, when the circumstances are correct, the tuplet's `nextElement()` function returns a note (or maybe a chord) and the note's (or chord's) `nextElement()` function returns a tuplet, etc.

**Comment by codecruisedor** [comment]
Understood, thanks for the extensive explanation. Working on it..I'll ask If I have any doubts and will create a PR once done.

**Comment by codecruisedor** [comment]
Hey @shoogle, perhaps this question stems due to my unfamiliarity with Western music but should this logic also apply to quadruples and other lengthy note clusters?
As in, the note's nextElement() function should return a cluster of say 'n' length. n would be 3 if it's a tuplet, 4 if it's a quadruple. Is my understanding correct?

**Comment by codecruisedor** [comment]
Thank you for answering...I have another query regarding this..

I can see that the Tuplet has a shape() method that seems to return the bracket object. 
So should we return the bracket in the note's nextElement() function(if elementType::TUPLET) or the tuplet itself? I can't see a method inside of the Tuplet class that returns the whole of the Tuplet..

And if we return the shape of the tuplet..then we would need to override nextElement() method for shape as well right?

**Comment by MarcSabatella** [comment]
You don't need a special method with the Tuplet class.  If you have an object of type Tuplet, you're done.  You don't need a method at all - it's the Tuplet object itself you want, not something returned by a method within the class.

**Comment by shoogle** [comment]
You'll end up with something that looks a little bit like this:

```C++
// chordrest.cpp
EngravingItem* ChordRest::nextElement()
{
    if (tuplet = nextChordRest(this)->tuplet()) {
        return tuplet;
    }
    // otherwise do whatever this function was already doing
}

// tuplet.cpp
EngravingItem* Tuplet::nextElement()
{
    return elements().front();
}
```

**Comment by codecruisedor** [comment]
I am starting to understand the code better now..so far, here are some of my observations, please clarify if anything is inaccurate.

-> Most engraving items have a nextElement() function (at least those items that are selectable ). In the method, they first return an engraving item from element() function. ` EngravingItem* e = score()->selection().element();` 
-> So I can determine the type of the element, and if its a tuple, just return it..like this..`if (e->type() == ElementType::TUPLET) {
        return e;
    }`
-> And in order to return the first element of the tuple, once the tuple is selected..I can return elements().front() as stated in the last comment
 `EngravingItem* Tuplet::nextElement()
{
    return elements().front();
}`
Are these valid?

**Comment by codecruisedor** [comment]
Have made a pull request for this issue : [#18346 ].

**Comment by brandonDavis264** [comment]
Can I help work on this issue?

**Comment by shoogle** [comment]
@brandonDavis264, sure, go ahead!

**Comment by efu98** [comment]
Hello! I can try to work on this :)

**Comment by wizofaus** [comment]
This would be great to get in, I currently have a score where I can't select triplets at all except by creating another one and using "select similar".
